Geometry/HcalCommonData/src/HcalGeomParameters.cc

```diff
@@ -285,58 +239,303 @@ void HcalGeomParameters::loadGeometry(const DDFilteredView& _fv, HcalParameters&
+            php.layHB.emplace_back(lay);
+          }
+        }
+      }
+      if (lay == 2) {
+        int iz = copy[nsiz - 5];
+        int fi = copy[nsiz - 4];
+        unsigned int it1 = find(iz, izb_);
+        if (it1 == izb_.size())
+          izb_.emplace_back(iz);
+        unsigned int it2 = find(fi, phib_);
+        if (it2 == phib_.size())
+          phib_.emplace_back(fi);
+      }
+      if (lay == 18) {
+        int ifi = -1, ich = -1;
+        if (nsiz > 2)
+          ifi = copy[nsiz - 3];
+        if (nsiz > 3)
+          ich = copy[nsiz - 4];
+        double z1 = std::abs((t.z()) + dz);
+        double z2 = std::abs((t.z()) - dz);
+        if (std::abs(z1 - z2) < 0.01)
+          z1 = 0;
+        if (ifi == 1 && ich == 4) {
+          if (z1 > z2) {
+            double tmp = z1;
+            z1 = z2;
+            z2 = tmp;
+          }
+          bool sok = true;
+          for (unsigned int kk = 0; kk < php.zHO.size(); kk++) {
+            if (std::abs(z2 - php.zHO[kk]) < 0.01) {
+              sok = false;
+              break;
+            } else if (z2 < php.zHO[kk]) {
+              php.zHO.resize(php.zHO.size() + 2);
+              for (unsigned int kz = php.zHO.size() - 1; kz > kk + 1; kz = kz - 2) {
+                php.zHO[kz] = php.zHO[kz - 2];
+                php.zHO[kz - 1] = php.zHO[kz - 3];
+              }
+              php.zHO[kk + 1] = z2;
+              php.zHO[kk] = z1;
+              sok = false;
+              break;
+            }
+          }
+          if (sok) {
+            php.zHO.emplace_back(z1);
+            php.zHO.emplace_back(z2);
+          }
+#ifdef EDM_ML_DEBUG
+          edm::LogVerbatim("HCalGeom") << "Detector " << idet << " Lay " << lay << " fi " << ifi << " " << ich << " z "
+                                       << z1 << " " << z2;
+#endif
+        }
+      }
+    } else if (idet == 4) {
+      // HE
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HCalGeom") << "HE " << fv.name() << " Shape "  //<< fv.shape()
+                                   << " Layer " << lay << " Z " << t.z();
+#endif
+      if (lay >= 0 && lay < 20) {
+        ie_[lay]++;
+        ze_[lay] += std::abs(t.z());
+        if (thke_[lay] <= 0)
+          thke_[lay] = dz;
+        double rinHE = t.Rho() * cos(alp) - dy;
+        double routHE = t.Rho() * cos(alp) + dy;
+        rminHE_[lay] += rinHE;
+        rmaxHE_[lay] += routHE;
+        bool found = false;
+        for (double k : php.zxHE) {
+          if (std::abs(k - std::abs(t.z())) < 0.01) {
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          php.zxHE.emplace_back(std::abs(t.z()));
+          php.rhoxHE.emplace_back(t.Rho() * std::cos(t.phi()));
+          php.dyHE.emplace_back(dy * std::cos(t.phi()));
+          dx1 -= 0.5 * (t.Rho() - dy) * std::cos(t.phi()) * std::tan(10 * CLHEP::deg);
+          dx2 -= 0.5 * (t.Rho() + dy) * std::cos(t.phi()) * std::tan(10 * CLHEP::deg);
```

Comment:
tan(10 degrees) should be made into a constant:
```
static const double tan10deg = std::tan(10._deg);
```
We'd like to reduce usage of CLHEP.